openssl/src/x509/extension.rs

```diff
@@ -506,14 +511,64 @@ impl SubjectAlternativeName {
 
     /// Sets the `otherName` flag.
     ///
-    /// Not currently actually supported, always panics.
-    #[deprecated = "other_name is deprecated and always panics. Please file a bug if you have a use case for this."]
+    /// Not currently actually supported, always panics. Please use other_name2 or other_name_ia5string
+    #[deprecated = "other_name is deprecated and always panics. Please use other_name2 or other_name_ia5string."]
     pub fn other_name(&mut self, _other_name: &str) -> &mut SubjectAlternativeName {
         unimplemented!(
-            "This has not yet been adapted for the new internals. File a bug if you need this."
+            "This has not yet been adapted for the new internals. Use other_name2 or other_name_ia5string."
         );
     }
 
+    /// Sets the `otherName` flag.
+    ///
+    /// `content` must be a valid der encoded ASN1_TYPE
+    ///
+    /// If you want to add just a ia5string use `other_name_ia5string`
+    pub fn other_name2(
+        &mut self,
+        oid: &str,
+        content: &[u8],
+    ) -> Result<&mut SubjectAlternativeName, ErrorStack> {
+        let typ: *mut ASN1_TYPE;
+        unsafe {
+            ffi::init();
+
+            typ = cvt_p(ffi::d2i_ASN1_TYPE(
+                ptr::null_mut(),
+                &mut content.as_ptr().cast(),
+                content.len().try_into().unwrap(),
+            ))?;
+        }
+        self.items
+            .push(RustGeneralName::OtherName(oid.to_string(), typ));
+        Ok(self)
+    }
```

Comment:
Would like @sfackler's take on this, specifically whether we should have this return a `Result`, or if it should be consistent with the other ones and just return `&mut SAN` and then do the error handling lazily.